generate-errors/src/lib.rs

```diff
@@ -1,114 +1,213 @@
+use anyhow::anyhow;
 use regex::Regex;
-use serde::Serialize;
-use std::fs::read_to_string;
-use std::{fs, io, path::PathBuf, str::FromStr};
-
-#[derive(Debug, Clone)]
-pub struct Section {
-    pub name: String,
-    pub content: Vec<ErrorCode>,
-    pub template: Option<String>,
-    pub header: Option<String>,
-    pub order: Option<usize>,
-    pub sort_order_reversed: bool,
-}
+use std::{
+    collections::HashMap,
+    fs,
+    path::{Path, PathBuf},
+};
 
-#[derive(Debug, Clone)]
-pub struct ErrorCode {
-    pub code: String,
-    pub content: String,
-}
+/// Gets the unordered content of the error pages
+/// supplied by user via
+/// a path to the directory containing
+/// the original Bevy error files.
+pub fn get_error_pages(bevy_errors_path: &Path) -> anyhow::Result<HashMap<String, String>> {
+    // Guard clause that determines
+    // if a path exists and is valid
+    // and not a broken symbolic link.
+    // Otherwise it errors and returns.
+    if !bevy_errors_path.try_exists()? {
+        return Err(anyhow!("The path ({bevy_errors_path:?}) is invalid"));
+    }
 
-#[derive(Serialize)]
-pub struct FrontMatterErrorCode {
-    pub title: String,
-    pub weight: usize,
-}
+    let mut error_page_paths: Vec<PathBuf> = vec![];
+
+    let entries = bevy_errors_path.read_dir()?;
+    let regex = Regex::new(r"B[0-9]{4}")?;
```

Comment:
```suggestion
    // Matches Bevy error codes, such as B0001, B0002, etc.
    let regex = Regex::new(r"B[0-9]{4}")?;
```

Regex can be confusing, especially to those who have not worked with it before. A small comment can help with that.